src/extension-runners/firefox-android.js

```diff
@@ -384,6 +387,29 @@ export class FirefoxAndroidExtensionRunner {
     await adbUtils.amForceStopAPK(selectedAdbDevice, selectedFirefoxApk);
   }
 
+  async adbArtifactsDir(removeArtifacts?: boolean) {
+    const {
+      adbUtils,
+      selectedAdbDevice,
+    } = this;
+
+    const foundDirectories = await adbUtils.checkOrCleanArtifacts(
+      selectedAdbDevice, removeArtifacts
+    );
+
+    if (foundDirectories) {
+      if (removeArtifacts) {
```

Comment:
we can avoid this nested if/else blocks, if I recall correctly in my previous review comment there were some pointers about how to achieve that.